_posts/2020-01-29-#17487.md

```diff
@@ -0,0 +1,115 @@
+---
+layout: pr
+date: 2020-01-29
+title: "Coins: allow Flush() without cache drop"
+pr: 17487
+authors: [jamesob]
+components: ["utxo db and indexes"]
+host: jamesob
+status: upcoming
+---
+
+## Notes
+
+The UTXO cache is a critical datastructure in Bitcoin for both correctness and
+performance. It's responsible for maintaining a view of the spendable coins
+based upon the transactions in blocks. The UTXO cache is just a denormalization
+(i.e. a copy of) of the information found in blocks, but it is often consulted
+exclusively for various things for performance reasons. For example while
```

Comment:
I think it's useful to think of the UTXO set as something wholly separate from the blockchain. For example, for a pruned node, the UTXO set isn't consulted for performance reasons. It's consulted because that's the place where the UTXOs are, and the block/transaction data might not even be available any more.